DREAM3DReviewFilters/CreateArrayofIndices.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Your License or Copyright can go here
+ */
+
+#include "CreateArrayofIndices.h"
+
+#include <numeric>
+
+#include <QtCore/QTextStream>
+
+#include "SIMPLib/Common/Constants.h"
+#include "SIMPLib/FilterParameters/DataArrayCreationFilterParameter.h"
+
+#include "DREAM3DReview/DREAM3DReviewConstants.h"
+#include "DREAM3DReview/DREAM3DReviewVersion.h"
+#include "SIMPLib/DataContainers/DataContainer.h"
+#include "SIMPLib/DataContainers/DataContainerArray.h"
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+CreateArrayofIndices::CreateArrayofIndices()
+: m_IndexArrayPath("IndexArray")
+{
+  initialize();
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+CreateArrayofIndices::~CreateArrayofIndices() = default;
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void CreateArrayofIndices::initialize()
+{
+  clearErrorCode();
+  clearWarningCode();
+  setCancel(false);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void CreateArrayofIndices::setupFilterParameters()
+{
+  FilterParameterVectorType parameters;
+  DataArrayCreationFilterParameter::RequirementType dacReq;
+  parameters.push_back(SIMPL_NEW_DA_CREATION_FP("Index Array Path", IndexArrayPath, FilterParameter::CreatedArray, CreateArrayofIndices, dacReq));
+  setFilterParameters(parameters);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void CreateArrayofIndices::dataCheck()
+{
+  clearErrorCode();
+  clearWarningCode();
+
+  std::vector<size_t> cDims(1, 1);
+  m_IndicesPtr = getDataContainerArray()->createNonPrereqArrayFromPath<DataArray<size_t>>(this, m_IndexArrayPath, 0, cDims);
+  if(nullptr != m_IndicesPtr.lock())
+  {
+    m_Indices = m_IndicesPtr.lock()->getPointer(0);
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void CreateArrayofIndices::execute()
+{
+  initialize();
+  dataCheck();
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  if(getCancel())
+  {
+    return;
+  }
+
+  SizeTArrayType& array = *(m_IndicesPtr.lock());
+  std::iota(array.begin(), array.end(), 0);
```

Comment:
```suggestion
  SizeTArrayType::Pointer indices = getDataContainerArray()->getPrereqArrayFromPath<SizeTArrayType>(this, m_IndexArrayPath, {1});
  if(indices == nullptr)
  {
    setErrorCondition(-99999, "Failed to obtain Indices DataArray");
    return;
  }

  std::iota(indices->begin(), indices->end(), static_cast<size_t>(0));
```

Now we try to avoid storing pointers to DataArrays in filters, and instead obtaining them in execute().